src/lib/blockchain_snark/blockchain_snark_state.ml

```diff
@@ -74,36 +74,46 @@ module Make_update (T : Transaction_snark.Verification.S) = struct
         and supply_increase_is_zero =
           Currency.Amount.(equal_var supply_increase (var_of_t zero))
         in
-        let%bind nothing_changed =
-          Boolean.(ledger_hash_didn't_change && supply_increase_is_zero)
-        in
-        let%bind new_pending_coinbase_hash, deleted_stack =
+        let%bind new_pending_coinbase_hash, deleted_stack, no_coinbases_popped
+            =
           let%bind root_after_delete, deleted_stack =
             Pending_coinbase.Checked.pop_coinbases prev_pending_coinbase_root
               ~proof_emitted:(Boolean.not ledger_hash_didn't_change)
           in
+          let%bind prev_state_body_hash =
```

Comment:
I think this should get passed in instead of recomputing it (you can see the previous_state_hash is already passed in, and the state body hash is computed in the process of computing that)

Reply:
> Looks great, just a few comments. The one other thing was I remember we discussed computing some boolean that was previous getting "exists'd in" that controlled how we were pushing to the stack. Did that happen in this PR?

Are you referring to the `is_new_stack` boolean? I removed it. We had that because we were consing state hashes (now it's just the state body hashes) and so had to request the boolean to get the correct previous state hash